x-pack/plugins/lens/public/embeddable/embeddable.tsx

```diff
@@ -610,6 +612,30 @@ export class Embeddable
 
   private fullAttributes: LensSavedObjectAttributes | undefined;
 
+  private handleExternalUserMessage = (messages: UserMessage[]) => {
+    if (this.input.customBadgeMessages) {
+      // we need something else to better identify those errors
+      const messagesToHandle = messages.filter(
+        (message) =>
+          message.displayLocations.some((d) => d.id === 'embeddableBadge') &&
+          message.severity === 'error'
+      );
```

Comment:
I think this is something that should happen at consumer side?
I'm just two mind here about the `severity: "info"` one, but maybe is it ok to keep it.

Reply:
I spent some time on it, I don't have any strong opinions about it, but it is, at first glance, a complex condition for the consumer. Perhaps we could let users decide about the `severity` but encapsulate the `displayLocations` condition?

Reply:
Ok, it makes sense to filter by location at this point.